File: src/Backend/PSR.Application/Services/LookupService.cs
using PSR.Application.Common;
using PSR.Application.Interfaces;
using PSR.Domain;
using PSR.SharedKernel;

namespace PSR.Application.Services
{
    public class LookupService : ILookupService
    {
        public List<LookupItem> GetCountries() {
            return Country.List().Select(x => new LookupItem(x.Id, x.Name)).ToList();
        }
    }
}

File: src/PSR.Angular/projects/corelib/src/lib/services/lookup.service.ts
import { HttpClient } from '@angular/common/http';
import { Inject, Injectable } from '@angular/core';
import { Observable } from 'rxjs';

import { AppEndpoints } from '../app.constants';
import { ILookupItem } from '../app.interfaces';
import { APP_CONFIG_TOKEN, IAppConfig } from '../tokens';

export interface Country {
    id: number;
    name: string;
}

@Injectable({
    providedIn: 'root'
})
export class LookupService {

    constructor(private http: HttpClient, @Inject(APP_CONFIG_TOKEN) private appConfig: IAppConfig) { }

    public getCountries(): Observable<ILookupItem[]> {
        return this.http.get<ILookupItem[]>(this.appConfig.getUrl(AppEndpoints.GetCountries));
    }
}
